python/pyspark/tests.py

```diff
@@ -2381,6 +2382,32 @@ def test_startTime(self):
         with SparkContext() as sc:
             self.assertGreater(sc.startTime, 0)
 
+    def test_forbid_insecure_gateway(self):
+        # By default, we fail immediately if you try to create a SparkContext
+        # with an insecure gateway
+        gateway = _launch_gateway(insecure=True)
+        with self.assertRaises(Exception) as context:
```

Comment:
Could you also use `with QuietTest(self.sc):`?

Reply:
I don't think so, since I need to pass in the spark context for QuietTest, but the exception is during construction.  Or is there a way to use it I'm not seeing?

I will manually add a `try / finally` to do the equivalent thing in any case

Reply:
I think it should work if you nest the statements like this
```python
with QuietTest(self.sc):
    with self.assertRaises(Exception) as context:
```

Reply:
the problem isn't the nesting of the `with` statements, its that `self.sc` you're passing into `QuietTest`.  That doesn't exist in this situation.  The point of this test is to get a failure inside the creation of the `sc`, so there is nothing i can pass in here.

But, `QuietTest` doesn't actually care about the `sc`, it just wants a handle on the jvm, to set the log4j configuration.  My manual `try / finally` does the same thing from the handle to the jvm in the java gateway.

Reply:
ah, I see. Sounds good then, thanks